src/app/shared/components/button/component.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ChangeDetectionStrategy, Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {Observable, Subject} from 'rxjs';
+import {tap, throttleTime} from 'rxjs/operators';
+
+@Component({
+  selector: 'km-button',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ButtonComponent implements OnInit, OnDestroy {
+  @Input() icon: string;
+  @Input() label: string;
+  @Input() observable: Observable<any>;
```

Comment:
If generic approach (`T`) does not work, I'd use here `unknown` instead of `any`